lighthouse-plugin-ad-speed-insights/audits/ad-blocking-tasks.js

```diff
@@ -135,61 +181,59 @@ class AdBlockingTasks extends Audit {
         .filter((record) => record.resourceType === 'Script')
-        .map((record) => record.url);
-
-    /** @type {LH.Audit.Details.Table['items']} */
-    const blocking = [];
-    let longTaskIndex = 0;
-    for (const adNetworkReq of adNetworkReqs) {
-      for (; longTaskIndex < longTasks.length; longTaskIndex++) {
-        const longTask = longTasks[longTaskIndex];
-        // Handle cases without any overlap.
-        if (longTask.endTime < fixTime(adNetworkReq.startTime)) continue;
-        if (fixTime(adNetworkReq.endTime) < longTask.startTime) break;
-
-        // Check if longTask delayed NetworkRecord.
-        if (fixTime(adNetworkReq.responseReceivedTime) < longTask.endTime) {
-          let scriptUrl = longTask.event.args.data ?
-            longTask.event.args.data.url :
-            // @ts-ignore
-            longTask.attributableURLs.find((url) => scriptUrls.includes(url));
-          if (!scriptUrl) {
-            if (longTask.attributableURLs.length) {
-              scriptUrl = longTask.attributableURLs[0];
-            } else {
-              continue;
-            }
-          }
-
-          if (isGpt(new URL(scriptUrl))) {
-            continue;
-          }
-
-          const taskName = longTask.event.name || '';
-          const name = TASK_NAMES[taskName] ? TASK_NAMES[taskName] : taskName;
-
-          blocking.push({
-            name,
-            script: scriptUrl,
-            group: longTask.group.label,
-            duration: longTask.selfTime,
-            adReqBlocked: adNetworkReq.url,
-          });
-        }
+        .map((record) => record.url));
+    const longTasks = tasks.filter((t) => isLong(t, knownScripts));
+
+    // TODO(warrengm): End on ad load rather than ad request end.
+    const endTime = fixTime(Math.min(...adNetworkReqs.map((r) => r.endTime)));
+
+    let blocking = [];
+    for (const longTask of longTasks) {
+      // Handle cases without any overlap.
+      if (longTask.startTime > endTime) {
+        continue;
+      }
+      const scriptUrl = attributableUrl(longTask, knownScripts);
+      if (scriptUrl && isGpt(new URL(scriptUrl))) {
+        continue;
       }
+
+      const taskName = longTask.event.name || '';
+      const name = TASK_NAMES[taskName] || taskName;
+
+      const url = scriptUrl && new URL(scriptUrl);
+      const displayUrl = url && (url.host + url.pathname);
+
+      blocking.push({
+        name,
+        script: displayUrl,
+        startTime: longTask.startTime,
+        endTime: longTask.endTime,
+        duration: longTask.duration,
+        isTopLevel: !longTask.parent,
+      });
+    }
+
+    const taskLimit = 10;
+    if (blocking.length > taskLimit) {
+      // For the sake of brevity, we show at most 5 long tasks. If needed we
+      // will filter tasks that are less actionable (child tasks or ones missing
+      // attributable URLs).
+      blocking = blocking.filter((b) => b.script && b.isTopLevel)
```

Comment:
By filtering out child tasks here, two runs, one with 10 long tasks and another with 11, could differ in the inclusion of a very long child task. Perhaps there could be a higher "long task" threshold here for child tasks.

Reply:
SGTM for determinism, though I don't think that very long child tasks are necessarily the most actionable.